src/manage_nsfs/nc_master_key_manager.js

```diff
@@ -340,13 +351,29 @@ class NCMasterKeysManager {
      * @returns {Promise<Object>}
      */
     async decrypt_access_keys(account) {
-        const decrypted_access_keys = await P.all(_.map(account.access_keys, async access_keys => ({
+        const decrypted_access_keys_basic = await P.all(_.map(account.access_keys, async access_keys => ({
                 access_key: access_keys.access_key,
                 secret_key: await this.decrypt(access_keys.encrypted_secret_key, account.master_key_id)
         })));
+        // To have the freedom to change the access_keys object
+        // we will iterate the array and add the secret_key
+        // and remove the encrypted_secret_key
+        const decrypted_access_keys = [];
+        const size = account.access_keys?.length ?? 0;
+        for (let index = 0; index < size; index++) {
+            const access_keys = { ...account.access_keys[index] };
+            delete access_keys.encrypted_secret_key;
+            access_keys.secret_key = decrypted_access_keys_basic[index].secret_key;
+            decrypted_access_keys[index] = access_keys;
+        }
```

Comment:
@shirady why another iteration is needed? can't we just put it inside P.all() line?
```
const decrypted_access_keys = await P.all(_.map(account.access_keys, async access_keys => {
      const decrypted_access_keys = { ...account.access_keys };
      decrypted_access_keys.secret_key = await this.decrypt(access_keys.encrypted_secret_key, 
                  account.master_key_id);
      delete decrypted_access_keys.encrypted_secret_key;
      return decrypted_access_keys;
}));

Reply:
I didn't find the way yet...
I want to avoid adding the properties directly in the first loop, and when I tried to move it all to a separate function I couldn't use the `this`.

Reply:
for what reason you want to avoid it?

Reply:
I want to avoid adding the properties directly in the first loop for 2 reasons:
1. In case it is `undefined` (the account doesn't have these properties) I do not want to set the key with `undefined` (if I do I would need to omit those properties).
2. To let us the flexibility to add properties in the schema without changing these functions again.

Reply:
@shirady 
1. just skip the description if secret_key and access_key are not defined
2. this row will do the flexibility - `const decrypted_access_keys = { ...account.access_keys };`

```
const decrypted_access_keys = await P.all(_.map(account.access_keys, async access_keys => {
      const decrypted_access_keys = { ...account.access_keys };
      if (decrypted_access_keys.encrypted_secret_key) {
            decrypted_access_keys.secret_key = await this.decrypt(access_keys.encrypted_secret_key, 
                   account.master_key_id);
            delete decrypted_access_keys.encrypted_secret_key;
      }
      return decrypted_access_keys;
}));
```

Reply:
I will use it, thank you.

Reply:
I will note that the `encrypted_secret_key` will appear at the end of the printed access object when looking at the config (for humans).